File: auto-pay-sdks/java-sdk/src/main/java/com/autopay/sdk/model/request/CreatePaymentRequest.java
package com.autopay.sdk.model.request;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.math.BigDecimal;
import java.util.HashMap;
import java.util.Map;

/**
 * 创建支付请求模型
 */
public class CreatePaymentRequest {
    
    @JsonProperty("order_id")
    private String orderId;
    
    @JsonProperty("amount")
    private BigDecimal amount;
    
    @JsonProperty("currency")
    private String currency;
    
    @JsonProperty("channel")
    private String channel;
    
    @JsonProperty("description")
    private String description;
    
    @JsonProperty("notify_url")
    private String notifyUrl;
    
    @JsonProperty("return_url")
    private String returnUrl;
    
    @JsonProperty("client_ip")
    private String clientIp;
    
    @JsonProperty("extra")
    private Map<String, Object> extra;
    
    @JsonProperty("metadata")
    private Map<String, Object> metadata;
    
    @JsonProperty("timeout")
    private Integer timeout;
    
    @JsonProperty("payment_method")
    private String paymentMethod;
    
    @JsonProperty("customer_id")
    private String customerId;
    
    @JsonProperty("customer")
    private CustomerInfo customer;
    
    // 构造方法
    public CreatePaymentRequest() {
        this.extra = new HashMap<>();
        this.metadata = new HashMap<>();
    }
    
    public CreatePaymentRequest(String orderId, BigDecimal amount, String currency, String channel) {
        this();
        this.orderId = orderId;
        this.amount = amount;
        this.currency = currency;
        this.channel = channel;
    }
    
    // Builder模式
    public static Builder builder() {
        return new Builder();
    }
    
    public static class Builder {
        private CreatePaymentRequest request = new CreatePaymentRequest();
        
        public Builder orderId(String orderId) {
            request.orderId = orderId;
            return this;
        }
        
        public Builder amount(BigDecimal amount) {
            request.amount = amount;
            return this;
        }
        
        public Builder amount(double amount) {
            request.amount = BigDecimal.valueOf(amount);
            return this;
        }
        
        public Builder currency(String currency) {
            request.currency = currency;
            return this;
        }
        
        public Builder channel(String channel) {
            request.channel = channel;
            return this;
        }
        
        public Builder description(String description) {
            request.description = description;
            return this;
        }
        
        public Builder notifyUrl(String notifyUrl) {
            request.notifyUrl = notifyUrl;
            return this;
        }
        
        public Builder returnUrl(String returnUrl) {
            request.returnUrl = returnUrl;
            return this;
        }
        
        public Builder clientIp(String clientIp) {
            request.clientIp = clientIp;
            return this;
        }
        
        public Builder extra(String key, Object value) {
            request.extra.put(key, value);
            return this;
        }
        
        public Builder metadata(String key, Object value) {
            request.metadata.put(key, value);
            return this;
        }
        
        public Builder timeout(Integer timeout) {
            request.timeout = timeout;
            return this;
        }
        
        public Builder paymentMethod(String paymentMethod) {
            request.paymentMethod = paymentMethod;
            return this;
        }
        
        public Builder customerId(String customerId) {
            request.customerId = customerId;
            return this;
        }
        
        public Builder customer(CustomerInfo customer) {
            request.customer = customer;
            return this;
        }
        
        public CreatePaymentRequest build() {
            // 验证必需参数
            if (request.orderId == null || request.orderId.trim().isEmpty()) {
                throw new IllegalArgumentException("订单ID不能为空");
            }
            if (request.amount == null || request.amount.compareTo(BigDecimal.ZERO) <= 0) {
                throw new IllegalArgumentException("支付金额必须大于0");
            }
            if (request.currency == null || request.currency.trim().isEmpty()) {
                throw new IllegalArgumentException("货币代码不能为空");
            }
            if (request.channel == null || request.channel.trim().isEmpty()) {
                throw new IllegalArgumentException("支付渠道不能为空");
            }
            
            return request;
        }
    }
    
    /**
     * 客户信息内部类
     */
    public static class CustomerInfo {
        @JsonProperty("name")
        private String name;
        
        @JsonProperty("email")
        private String email;
        
        @JsonProperty("phone")
        private String phone;
        
        @JsonProperty("address")
        private String address;
        
        // 默认构造方法
        public CustomerInfo() {}
        
        public CustomerInfo(String name, String email) {
            this.name = name;
            this.email = email;
        }
        
        // Getter和Setter
        public String getName() {
            return name;
        }
        
        public void setName(String name) {
            this.name = name;
        }
        
        public String getEmail() {
            return email;
        }
        
        public void setEmail(String email) {
            this.email = email;
        }
        
        public String getPhone() {
            return phone;
        }
        
        public void setPhone(String phone) {
            this.phone = phone;
        }
        
        public String getAddress() {
            return address;
        }
        
        public void setAddress(String address) {
            this.address = address;
        }
        
        @Override
        public String toString() {
            return String.format("CustomerInfo{name='%s', email='%s', phone='%s'}", 
                    name, email, phone);
        }
    }
    
    // Getters and Setters
    public String getOrderId() {
        return orderId;
    }
    
    public void setOrderId(String orderId) {
        this.orderId = orderId;
    }
    
    public BigDecimal getAmount() {
        return amount;
    }
    
    public void setAmount(BigDecimal amount) {
        this.amount = amount;
    }
    
    public String getCurrency() {
        return currency;
    }
    
    public void setCurrency(String currency) {
        this.currency = currency;
    }
    
    public String getChannel() {
        return channel;
    }
    
    public void setChannel(String channel) {
        this.channel = channel;
    }
    
    public String getDescription() {
        return description;
    }
    
    public void setDescription(String description) {
        this.description = description;
    }
    
    public String getNotifyUrl() {
        return notifyUrl;
    }
    
    public void setNotifyUrl(String notifyUrl) {
        this.notifyUrl = notifyUrl;
    }
    
    public String getReturnUrl() {
        return returnUrl;
    }
    
    public void setReturnUrl(String returnUrl) {
        this.returnUrl = returnUrl;
    }
    
    public String getClientIp() {
        return clientIp;
    }
    
    public void setClientIp(String clientIp) {
        this.clientIp = clientIp;
    }
    
    public Map<String, Object> getExtra() {
        return extra;
    }
    
    public void setExtra(Map<String, Object> extra) {
        this.extra = extra;
    }
    
    public Map<String, Object> getMetadata() {
        return metadata;
    }
    
    public void setMetadata(Map<String, Object> metadata) {
        this.metadata = metadata;
    }
    
    public Integer getTimeout() {
        return timeout;
    }
    
    public void setTimeout(Integer timeout) {
        this.timeout = timeout;
    }
    
    public String getPaymentMethod() {
        return paymentMethod;
    }
    
    public void setPaymentMethod(String paymentMethod) {
        this.paymentMethod = paymentMethod;
    }
    
    public String getCustomerId() {
        return customerId;
    }
    
    public void setCustomerId(String customerId) {
        this.customerId = customerId;
    }
    
    public CustomerInfo getCustomer() {
        return customer;
    }
    
    public void setCustomer(CustomerInfo customer) {
        this.customer = customer;
    }
    
    @Override
    public String toString() {
        return "CreatePaymentRequest{" +
                "orderId='" + orderId + '\'' +
                ", amount=" + amount +
                ", currency='" + currency + '\'' +
                ", channel='" + channel + '\'' +
                ", description='" + description + '\'' +
                ", notifyUrl='" + notifyUrl + '\'' +
                ", returnUrl='" + returnUrl + '\'' +
                ", clientIp='" + clientIp + '\'' +
                ", extra=" + extra +
                ", metadata=" + metadata +
                ", timeout=" + timeout +
                ", paymentMethod='" + paymentMethod + '\'' +
                ", customerId='" + customerId + '\'' +
                ", customer=" + customer +
                '}';
    }
}

File: auto-pay-sdks/nodejs-sdk/src/models/CreatePaymentRequest.ts
import { Currency, PaymentChannel, CustomerInfo, Metadata } from '../types';

/**
 * 创建支付请求
 */
export class CreatePaymentRequest {
  private readonly _orderId: string;
  private readonly _amount: number;
  private readonly _currency: Currency;
  private readonly _channel: PaymentChannel;
  private readonly _description?: string;
  private readonly _subject?: string;
  private readonly _body?: string;
  private readonly _notifyUrl?: string;
  private readonly _returnUrl?: string;
  private readonly _clientIp?: string;
  private readonly _customerId?: string;
  private readonly _customer?: CustomerInfo;
  private readonly _metadata?: Metadata;
  private readonly _timeout?: number;
  private readonly _extra?: Record<string, any>;

  public constructor(builder: CreatePaymentRequestBuilder) {
    this._orderId = builder.orderId!;
    this._amount = builder.amount!;
    this._currency = builder.currency!;
    this._channel = builder.channel!;
    this._description = builder.description;
    this._subject = builder.subject;
    this._body = builder.body;
    this._notifyUrl = builder.notifyUrl;
    this._returnUrl = builder.returnUrl;
    this._clientIp = builder.clientIp;
    this._customerId = builder.customerId;
    this._customer = builder.customer;
    this._metadata = builder.metadata;
    this._timeout = builder.timeout;
    this._extra = builder.extra;
  }

  /**
   * 创建构建器
   */
  public static builder(): CreatePaymentRequestBuilder {
    return new CreatePaymentRequestBuilder();
  }

  /**
   * 获取订单ID
   */
  public get orderId(): string {
    return this._orderId;
  }

  /**
   * 获取支付金额
   */
  public get amount(): number {
    return this._amount;
  }

  /**
   * 获取货币代码
   */
  public get currency(): Currency {
    return this._currency;
  }

  /**
   * 获取支付渠道
   */
  public get channel(): PaymentChannel {
    return this._channel;
  }

  /**
   * 获取订单描述
   */
  public get description(): string | undefined {
    return this._description;
  }

  /**
   * 获取订单标题
   */
  public get subject(): string | undefined {
    return this._subject;
  }

  /**
   * 获取订单内容
   */
  public get body(): string | undefined {
    return this._body;
  }

  /**
   * 获取异步通知URL
   */
  public get notifyUrl(): string | undefined {
    return this._notifyUrl;
  }

  /**
   * 获取同步返回URL
   */
  public get returnUrl(): string | undefined {
    return this._returnUrl;
  }

  /**
   * 获取客户端IP
   */
  public get clientIp(): string | undefined {
    return this._clientIp;
  }

  /**
   * 获取客户ID
   */
  public get customerId(): string | undefined {
    return this._customerId;
  }

  /**
   * 获取客户信息
   */
  public get customer(): CustomerInfo | undefined {
    return this._customer;
  }

  /**
   * 获取元数据
   */
  public get metadata(): Metadata | undefined {
    return this._metadata;
  }

  /**
   * 获取超时时间（秒）
   */
  public get timeout(): number | undefined {
    return this._timeout;
  }

  /**
   * 获取额外参数
   */
  public get extra(): Record<string, any> | undefined {
    return this._extra;
  }

  /**
   * 转换为JSON对象
   */
  public toJSON(): Record<string, any> {
    return {
      order_id: this._orderId,
      amount: this._amount,
      currency: this._currency,
      channel: this._channel,
      description: this._description,
      subject: this._subject,
      body: this._body,
      notify_url: this._notifyUrl,
      return_url: this._returnUrl,
      client_ip: this._clientIp,
      customer_id: this._customerId,
      customer: this._customer,
      metadata: this._metadata,
      timeout: this._timeout,
      ...this._extra
    };
  }
}

/**
 * 创建支付请求构建器
 */
export class CreatePaymentRequestBuilder {
  private _orderId?: string;
  private _amount?: number;
  private _currency?: Currency;
  private _channel?: PaymentChannel;
  private _description?: string;
  private _subject?: string;
  private _body?: string;
  private _notifyUrl?: string;
  private _returnUrl?: string;
  private _clientIp?: string;
  private _customerId?: string;
  private _customer?: CustomerInfo;
  private _metadata?: Metadata;
  private _timeout?: number;
  private _extra?: Record<string, any>;

  /**
   * 获取/设置订单ID
   */
  public get orderId(): string | undefined {
    return this._orderId;
  }

  public set orderId(value: string) {
    this._orderId = value;
  }

  /**
   * 获取/设置支付金额
   */
  public get amount(): number | undefined {
    return this._amount;
  }

  public set amount(value: number) {
    this._amount = value;
  }

  /**
   * 获取/设置货币代码
   */
  public get currency(): Currency | undefined {
    return this._currency;
  }

  public set currency(value: Currency) {
    this._currency = value;
  }

  /**
   * 获取/设置支付渠道
   */
  public get channel(): PaymentChannel | undefined {
    return this._channel;
  }

  public set channel(value: PaymentChannel) {
    this._channel = value;
  }

  /**
   * 获取/设置订单描述
   */
  public get description(): string | undefined {
    return this._description;
  }

  public set description(value: string) {
    this._description = value;
  }

  /**
   * 获取/设置订单标题
   */
  public get subject(): string | undefined {
    return this._subject;
  }

  public set subject(value: string) {
    this._subject = value;
  }

  /**
   * 获取/设置订单内容
   */
  public get body(): string | undefined {
    return this._body;
  }

  public set body(value: string) {
    this._body = value;
  }

  /**
   * 获取/设置异步通知URL
   */
  public get notifyUrl(): string | undefined {
    return this._notifyUrl;
  }

  public set notifyUrl(value: string) {
    this._notifyUrl = value;
  }

  /**
   * 获取/设置同步返回URL
   */
  public get returnUrl(): string | undefined {
    return this._returnUrl;
  }

  public set returnUrl(value: string) {
    this._returnUrl = value;
  }

  /**
   * 获取/设置客户端IP
   */
  public get clientIp(): string | undefined {
    return this._clientIp;
  }

  public set clientIp(value: string) {
    this._clientIp = value;
  }

  /**
   * 获取/设置客户ID
   */
  public get customerId(): string | undefined {
    return this._customerId;
  }

  public set customerId(value: string) {
    this._customerId = value;
  }

  /**
   * 获取/设置客户信息
   */
  public get customer(): CustomerInfo | undefined {
    return this._customer;
  }

  public set customer(value: CustomerInfo) {
    this._customer = value;
  }

  /**
   * 获取/设置元数据
   */
  public get metadata(): Metadata | undefined {
    return this._metadata;
  }

  public set metadata(value: Metadata) {
    this._metadata = value;
  }

  /**
   * 获取/设置超时时间（秒）
   */
  public get timeout(): number | undefined {
    return this._timeout;
  }

  public set timeout(value: number) {
    this._timeout = value;
  }

  /**
   * 获取/设置额外参数
   */
  public get extra(): Record<string, any> | undefined {
    return this._extra;
  }

  public set extra(value: Record<string, any>) {
    this._extra = value;
  }

  /**
   * 设置订单ID
   */
  public setOrderId(orderId: string): CreatePaymentRequestBuilder {
    this.orderId = orderId;
    return this;
  }

  /**
   * 设置支付金额
   */
  public setAmount(amount: number): CreatePaymentRequestBuilder {
    this.amount = amount;
    return this;
  }

  /**
   * 设置货币代码
   */
  public setCurrency(currency: Currency): CreatePaymentRequestBuilder {
    this.currency = currency;
    return this;
  }

  /**
   * 设置支付渠道
   */
  public setChannel(channel: PaymentChannel): CreatePaymentRequestBuilder {
    this.channel = channel;
    return this;
  }

  /**
   * 设置订单描述
   */
  public setDescription(description: string): CreatePaymentRequestBuilder {
    this.description = description;
    return this;
  }

  /**
   * 设置订单标题
   */
  public setSubject(subject: string): CreatePaymentRequestBuilder {
    this.subject = subject;
    return this;
  }

  /**
   * 设置订单内容
   */
  public setBody(body: string): CreatePaymentRequestBuilder {
    this.body = body;
    return this;
  }

  /**
   * 设置异步通知URL
   */
  public setNotifyUrl(notifyUrl: string): CreatePaymentRequestBuilder {
    this.notifyUrl = notifyUrl;
    return this;
  }

  /**
   * 设置同步返回URL
   */
  public setReturnUrl(returnUrl: string): CreatePaymentRequestBuilder {
    this.returnUrl = returnUrl;
    return this;
  }

  /**
   * 设置客户端IP
   */
  public setClientIp(clientIp: string): CreatePaymentRequestBuilder {
    this.clientIp = clientIp;
    return this;
  }

  /**
   * 设置客户ID
   */
  public setCustomerId(customerId: string): CreatePaymentRequestBuilder {
    this.customerId = customerId;
    return this;
  }

  /**
   * 设置客户信息
   */
  public setCustomer(customer: CustomerInfo): CreatePaymentRequestBuilder {
    this.customer = customer;
    return this;
  }

  /**
   * 设置元数据
   */
  public setMetadata(metadata: Metadata): CreatePaymentRequestBuilder {
    this.metadata = metadata;
    return this;
  }

  /**
   * 添加元数据项
   */
  public addMetadata(key: string, value: any): CreatePaymentRequestBuilder {
    if (!this._metadata) {
      this._metadata = {};
    }
    this._metadata[key] = value;
    return this;
  }

  /**
   * 设置超时时间（秒）
   */
  public setTimeout(timeout: number): CreatePaymentRequestBuilder {
    this.timeout = timeout;
    return this;
  }

  /**
   * 设置额外参数
   */
  public setExtra(extra: Record<string, any>): CreatePaymentRequestBuilder {
    this.extra = extra;
    return this;
  }

  /**
   * 添加额外参数
   */
  public addExtra(key: string, value: any): CreatePaymentRequestBuilder {
    if (!this._extra) {
      this._extra = {};
    }
    this._extra[key] = value;
    return this;
  }

  /**
   * 验证参数
   */
  private validate(): void {
    if (!this._orderId || this._orderId.trim() === '') {
      throw new Error('订单ID不能为空');
    }

    if (!this._amount || this._amount <= 0) {
      throw new Error('支付金额必须大于0');
    }

    if (!this._currency) {
      throw new Error('货币代码不能为空');
    }

    if (!this._channel) {
      throw new Error('支付渠道不能为空');
    }
  }

  /**
   * 构建请求对象
   */
  public build(): CreatePaymentRequest {
    this.validate();
    return new CreatePaymentRequest(this);
  }
}